### PR TITLE
Feature/permissions

### PR DESCRIPTION
**Pull request by victoriiazaporozhan**
## PR Checklist

* [x] Have you verified that the PR is pointing to the correct target branch? (`develop` for features/bugfixes, other if mentioned in the task)
* [x] Have you verified that your branch is consistent with the target branch and has no conflicts? (if not, make a rebase under the target branch)
* [x] Have you checked that everything works within the branch according to the task description and tested it locally?
* [x] Have you run the linter (`npm run lint`) prior to submission? Enable the git hook on commit in your IDE to run it and format the code automatically.
* [x] Have you run the tests locally and added/updated them if needed?
* [x] Have you checked that app can be built (`npm run build`)?
* [x] Have you checked that no new circular dependencies appreared with your changes? (the webpack plugin reports circular dependencies within the `dev` npm script)
* [x] Have you made sure that all the necessary pipelines has been successfully completed?
* [x] If the task requires translations to be updated, have you done this by running the `manage:translations` script?
* [x] Have you added the link to the PR in the Jira ticket comments?

## Summary by CodeRabbit

- New Features
  - Granular, role-based UI: buttons and menus now appear only when permitted (test cases: create/import/edit/move/duplicate/delete, add to launch/plan; test plans: create/edit/duplicate/delete; manual launches; filters/widgets; BTS/defects; sidebars and organization/project tools). Empty states adapt accordingly.
- Bug Fixes
  - Added missing React key to social posts; removed empty menus when no permitted actions.
- Refactor
  - Centralized permission checks via a new hook; unified role selection.
- Style
  - Adjusted padding in test case list rows.
- Chores
  - Added “hooks” import alias for simpler module resolution.

### CODE REVIEW

File: app/src/pages/inside/testCaseLibraryPage/testCaseList/testCaseSidePanel/testCaseSidePanel.tsx [hunk 1]
```diff
@@ -106,7 +116,18 @@ export const TestCaseSidePanel = memo(
...
+    const getExcludedActions = () => {
+      const excludedActions: TestCaseMenuAction[] = [];
+
+      if (!canEditTestCase) excludedActions.push(TestCaseMenuAction.EDIT);
+      if (!canDeleteTestCase) excludedActions.push(TestCaseMenuAction.DELETE);
+      if (!canDuplicateTestCase) excludedActions.push(TestCaseMenuAction.DUPLICATE);
+      if (!canMoveTestCase) excludedActions.push(TestCaseMenuAction.MOVE);
+
+      return excludedActions;
+    };
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/testCaseLibraryPage/testCaseList/testCaseSidePanel/testCaseSidePanel.tsx:128]: I see [similar code](https://github.com/reportportal/service-ui/pull/4578/files#diff-e6f21610e24e1a3df5fc2ecd00ed0ebfb0104e0bacf4790b6256e018e411d9cfR47) in the PR. We can remove duplication by creating a reusable function and passing a different `permissionMap`

```
type PermissionMapEntry = {
  isAllowed: boolean;
  action: TestCaseMenuAction;
};

const getExcludedActionsFromPermissionMap = (
  permissionMap: PermissionMapEntry[]
): TestCaseMenuAction[] => permissionMap
    .filter(({ isAllowed }) => !isAllowed)
    .map(({ action }) => action);

// Usage
const permissionMap = [
  { isAllowed: canEditTestCase, action: TestCaseMenuAction.EDIT },
  { isAllowed: canDeleteTestCase, action: TestCaseMenuAction.DELETE },
  { isAllowed: canDuplicateTestCase, action: TestCaseMenuAction.DUPLICATE },
  { isAllowed: canMoveTestCase, action: TestCaseMenuAction.MOVE },
];

const menuItems = createTestCaseMenuItems(formatMessage, {}, getExcludedActionsFromPermissionMap(permissionMap));
```

File: app/src/pages/inside/projectSettingsPageContainer/content/analyzerContainer/analyzerContainer.jsx [hunk 2]
```diff
@@ -68,12 +67,12 @@ export const AnalyzerContainer = ({ setHeaderNodes }) => {
...
   const analyzerConfig = useSelector(analyzerAttributesSelector);
   const analyzerExtensions = useSelector(analyzerExtensionsSelector);
-  const userRoles = useSelector(userRolesSelector);
+  const { canUpdateSettings } = useUserPermissions();
   const isAnalyzerServiceAvailable = !!analyzerExtensions.length;
   const analyzerUnavailableTitle = !isAnalyzerServiceAvailable
     ? formatMessage(COMMON_LOCALE_KEYS.ANALYZER_DISABLED)
     : null;
-  const hasPermission = canUpdateSettings(userRoles);
+  const hasPermission = canUpdateSettings;
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/projectSettingsPageContainer/content/analyzerContainer/analyzerContainer.jsx:75]: Let's remove `hasPermission` and use `canUpdateSettings` only

File: app/src/pages/inside/projectSettingsPageContainer/content/defectTypes/defectTypes.jsx [hunk 3]
```diff
@@ -100,7 +99,7 @@ export const DefectTypes = ({ setHeaderTitleNode }) => {
     () => DEFECT_TYPES_SEQUENCE.reduce((acc, groupName) => defectTypes[groupName]?.length + acc, 0),
     [defectTypes],
   );
-  const isEditable = canUpdateSettings(userRoles);
+  const isEditable = canUpdateSettings;
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/projectSettingsPageContainer/content/defectTypes/defectTypes.jsx:102]: Let's remove `isEditable` and use only `canUpdateSettings`

File: app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationInfo.jsx [hunk 4]
```diff
@@ -57,11 +56,11 @@ export const IntegrationInfo = (props) => {
   const { formatMessage } = useIntl();
   const { trackEvent } = useTracking();
   const settingsExtensions = useSelector(uiExtensionIntegrationSettingsSelector);
-  const userRoles = useSelector(userRolesSelector);
+  const { canUpdateSettings } = useUserPermissions();
   const globalIntegrations = useSelector(namedGlobalIntegrationsSelector);
   const projectIntegrations = useSelector(namedProjectIntegrationsSelector);
   const { organizationSlug, projectSlug } = useSelector(urlOrganizationAndProjectSelector);
-  const isAbleToClick = canUpdateSettings(userRoles);
+  const isAbleToClick = canUpdateSettings;
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationInfo/integrationInfo.jsx:63]: Let's remove `isAbleToClick` and use only `canUpdateSettings`

File: app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationsList.jsx [hunk 5]
```diff
@@ -35,8 +33,8 @@ export const IntegrationsList = (props) => {
   const { formatMessage } = useIntl();
   const { availableIntegrations, onItemClick } = props;
   const { trackEvent } = useTracking();
-  const userRoles = useSelector(userRolesSelector);
-  const isEditor = canUpdateSettings(userRoles);
+  const { canUpdateSettings } = useUserPermissions();
+  const isEditor = canUpdateSettings;
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/projectSettingsPageContainer/content/integrations/integrationsList/integrationsList.jsx:37]: Let's remove `isEditor` and use `canUpdateSettings` only

File: app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx [hunk 6]
```diff
@@ -56,11 +55,11 @@ export const Notifications = () => {
   const dispatch = useDispatch();
   const { trackEvent } = useTracking();
 
-  const userRoles = useSelector(userRolesSelector);
+  const { canUpdateSettings } = useUserPermissions();
   const isAllNotificationsEnabled = useSelector(projectNotificationsStateSelector);
   const notifications = useSelector(projectNotificationsSelector);
   const loading = useSelector(projectNotificationsLoadingSelector);
-  const isReadOnly = !canUpdateSettings(userRoles);
+  const isReadOnly = !canUpdateSettings;
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/projectSettingsPageContainer/content/notifications/notifications.jsx:62]: Let's remove `isReadOnly` and use `canUpdateSettings` only

File: app/src/pages/inside/testCaseLibraryPage/emptyState/details/detailsEmptyState.tsx [hunk 7]
```diff
@@ -18,25 +18,35 @@ import Parser from 'html-react-parser';
...
+    if (canEditTestCaseScenario) {
+      buttons.push({
+        isCompact: true,
+        name: formatMessage(messages.editScenario),
+        variant: 'primary',
+      });
+    }
+
+    return buttons;
+  };
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/testCaseLibraryPage/emptyState/details/detailsEmptyState.tsx:41]:
```suggestion
  const getActionButtons = () => canEditTestCaseScenario ? [{
        isCompact: true,
        name: formatMessage(messages.editScenario),
        variant: 'primary',
      }] : [];
```
you can also wrap it with `useCallback`

File: app/src/pages/inside/testCaseLibraryPage/emptyState/folder/folderEmptyState.tsx [hunk 8]
```diff
@@ -15,38 +15,59 @@
...
+  dataAutomationId: string;
+  icon: FC<SVGProps<SVGSVGElement>>;
+  variant?: string;
+  isCompact: boolean;
+}
 export const FolderEmptyState = () => {
   const { formatMessage } = useIntl();
+  const { canCreateTestCase, canImportTestCases } = useUserPermissions();
+
+  const availableButtons = () => {
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/testCaseLibraryPage/emptyState/folder/folderEmptyState.tsx:40]:
```suggestion
  const getAvailableButtons = () => {
```

File: app/src/pages/inside/testCaseLibraryPage/emptyState/folder/folderEmptyState.tsx [hunk 9]
```diff
@@ -15,38 +15,59 @@
...
 
 import { EmptyStatePage } from 'pages/inside/common/emptyStatePage';
 import ImportIcon from 'common/img/import-thin-inline.svg';
 import PlusIconInline from 'common/img/plus-button-inline.svg';
+import { useUserPermissions } from 'hooks/useUserPermissions';
 
 import { messages } from '../messages';
 import { commonMessages } from '../../commonMessages';
 
+interface Button {
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/testCaseLibraryPage/emptyState/folder/folderEmptyState.tsx:29]: Define this type once and use in all other similar places. For example I see here you defined it in [another way](https://github.com/reportportal/service-ui/pull/4578/files). There are other places too

File: app/src/pages/inside/testCaseLibraryPage/emptyState/mainPage/mainPageEmptyState.tsx [hunk 10]
```diff
@@ -49,28 +59,39 @@ export const MainPageEmptyState = () => {
     (translation) => Parser(formatMessage(translation, {}, { ignoreTag: true })),
   );
 
+  const availableButtons = () => {
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/testCaseLibraryPage/emptyState/mainPage/mainPageEmptyState.tsx:62]:
```suggestion
  const getAvailableButtons = () => {
```

File: app/src/pages/inside/testCaseLibraryPage/testCaseDetailsPage/testCaseDetailsHeader/testCaseDetailsHeader.tsx [hunk 11]
```diff
@@ -82,6 +91,27 @@ export const TestCaseDetailsHeader = ({
...
+      {
+        label: formatMessage(messages.historyOfActions),
+        onClick: handleHistoryOfActions,
+      },
+    ];
+
+    if (canDuplicateTestCase) {
+      items.push({ label: formatMessage(COMMON_LOCALE_KEYS.DUPLICATE) });
+    }
+    if (canDeleteTestCase) {
```
Comment (ViktorSoroka07) [inline app/src/pages/inside/testCaseLibraryPage/testCaseDetailsPage/testCaseDetailsHeader/testCaseDetailsHeader.tsx:105]:
```suggestion
    }

    if (canDeleteTestCase) {
```

File: app/src/pages/inside/testCaseLibraryPage/testCaseDetailsPage/testCaseDetailsHeader/testCaseDetailsHeader.tsx [hunk 12]
```diff
@@ -82,6 +91,27 @@ export const TestCaseDetailsHeader = ({
     });
   };
 
+  const menuItems = () => {
```
Comment (OleksandrDragun) [inline app/src/pages/inside/testCaseLibraryPage/testCaseDetailsPage/testCaseDetailsHeader/testCaseDetailsHeader.tsx:94]:
```suggestion
  const getMenuItems = () => {
```

File: app/src/pages/inside/testCaseLibraryPage/emptyState/details/detailsEmptyState.tsx (hunk 7 above)
Comment (victoriiazaporozhan) [reply app/src/pages/inside/testCaseLibraryPage/emptyState/details/detailsEmptyState.tsx:41]: @ViktorSoroka07 I think here is no need in useCallback as `canEditTestCaseScenario` always the same and won't cause unnecessary re-renders
